macros/union_data.sql

```diff
@@ -92,16 +92,38 @@
     {%- endif -%}
 
 {%- else -%}
-    {%- set relation=adapter.get_relation(
-        database=var(database_variable, default_database),
-        schema=var(schema_variable, default_schema),
-        identifier=var(default_schema ~ '_' ~ table_identifier ~ '_' ~ 'identifier', table_identifier)) -%}
-
-{%- set table_exists=relation is not none -%}
+    {% set exception_schemas = {"linkedin_company_pages": "linkedin_pages", "instagram_business_pages": "instagram_business"} %}
+    {% set relation = namespace(value="") %}
+    {% if default_schema in exception_schemas.keys() %}
+        {% for corrected_schema_name in exception_schemas.items() %}   
+            {% if default_schema in corrected_schema_name %}
+                {# In order for this macro to effectively work within upstream integration tests (mainly used by the Fivetran dbt package maintainers), this identifier variable selection is required to use the macro with different identifier names. #}
+                {% set identifier_var = corrected_schema_name[1] + "_" + table_identifier + "_identifier"  %}
+                {%- set relation.value=adapter.get_relation(
+                    database=source(corrected_schema_name[1], table_identifier).database,
+                    schema=source(corrected_schema_name[1], table_identifier).schema,
+                    identifier=var(identifier_var, table_identifier)
```

Comment:
this works for now, but this reallllly wouldn't work for something like the [apple store](https://github.com/fivetran/dbt_apple_store_source/blob/main/models/src_apple_store.yml), how do we feel about that?

Reply:
This "technically would" work for a package like apple store. We would just need to add apple store to the exception dictionary. I agree that this isn't a great fool proof solution, but for right now it provides the coverage needed to ensure the updates work as intended.